### ISSUE TITLE
Kong routes requests to targets deleted

### ISSUE THREAD

**Issue by altnti** [Kong/kong#10722]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

2.8.2

### Current Behavior

Delete target 10.55.7.35:50787  and add target 10.55.7.182:50787 for one upstream.
But the request maybe route to 10.55.7.35:50787.

Check the db:target, it only one entity:10.55.7.182:50787.

After I  try add 10.55.7.35:50787 and remove 10.55.7.35:50787 from admin api, but still  maybe route the request to 10.55.7.35:50787.

The Kong cluster have tow node. One of them has this problem.

### Expected Behavior

All nodes can be update correctly when operate the upstream from the Admin api.

### Steps To Reproduce

There is a probability that it cannot reproduce stably.

### Anything else?

Some configure:
```
worker_consistency = eventual
worker_state_update_frequency = 20
db_update_frequency = 5
db_update_propagation = 0
db_cache_ttl = 0
```
db: PGSql

I find some issue: [8757](https://github.com/Kong/kong/issues/8757)

**Comment by altnti** [comment]
This issue has reappeared recently. It is likely to occur when removing many of targets at the same time.

**Comment by hbagdi** [comment]
@locao Insights?

**Comment by locao** [comment]
Hi @altnti, thanks for your report. 

This issue was fixed in #10384 and was released in Kong 3.2.2. As we don't have plans for new Kong 2.x releases, the fix was not backported. 

If this issue is happening frequently, we recommend that you upgrade your Kong installation to the latest version. If you plan to stick with Kong 2.x longer, you can apply the fix in #10384 to the [`release/2.8.x` branch](https://github.com/Kong/kong/tree/release/2.8.x) and build Kong using [the supplied instructions in DEVELOPER.md file](https://github.com/Kong/kong/blob/master/DEVELOPER.md#build-and-install-from-source).

**Comment by altnti** [comment]
Awesome!  Thank you. @locao
